Realm/Realm/Linq/WhereClauseVisitor.cs

```diff
@@ -0,0 +1,252 @@
+                node.Method.DeclaringType == typeof(StringExtensions))
+            {
+                ComparisonNode result;
+                var test = node.Method.Name;
+                if (test.Equals("StartsWith"))
+                {
+                    result = new StartsWithNode();
+                }
+                else if (test.Equals("EndsWith"))
+                {
+                    result = new EndsWithNode();
+                }
+                else if (test.Equals("Contains"))
+                {
+                    result = new ContainsNode();
+                }
+
+                // TODO Undecided implementation
+                else if (test.Equals("Like"))
+                {
+                    result = new LikeNode();
+                }
+                else
+                {
+                    throw new NotSupportedException(node.Method.Name + " is not supported string operation method");
+                }
+
+                if (node.Object is MemberExpression me)
+                {
+                    if (me.Expression != null && me.Expression.NodeType == ExpressionType.Parameter)
+                    {
+                        result.Left = new PropertyNode();
+                        var leftName = GetColumnName(me, me.NodeType);
+                        result.Left.Name = leftName;
+                        result.Left.Type = GetKind(me.Type);
+                    }
+                    else
+                    {
+                        throw new NotSupportedException(me + " is null or not a supported type.");
+                    }
+
+                    if (node.Arguments[0] is ConstantExpression ce)
+                    {
+                        result.Right = new ConstantNode();
+                        result.Right.Value = ce.Value;
+                        result.Right.Type = GetKind(ce.Value.GetType());
+                    }
+                }
+
+                returnNode = result;
+            }
+            else
+            {
+                throw new NotSupportedException(node.Method + " is not a supported method.");
+            }
+
+            return RealmLinqExpression.Create(returnNode);
+        }
+
+        protected override Expression VisitBinary(BinaryExpression be)
+        {
+            ExpressionNode returnNode;
+            if (be.NodeType == ExpressionType.AndAlso)
+            {
+                var andNode = new AndNode();
+                andNode.Left = Extract(be.Left);
+                andNode.Right = Extract(be.Right);
+                returnNode = andNode;
+            }
+            else if (be.NodeType == ExpressionType.OrElse)
+            {
+                var orNode = new OrNode();
+                orNode.Left = Extract(be.Left);
+                orNode.Right = Extract(be.Right);
+                returnNode = orNode;
+            }
+            else
+            {
+                ComparisonNode comparisonNode;
+                switch (be.NodeType)
```

Comment:
When we bump the C# version, we can use the simplified switch expression here too